### PR TITLE
Add map functions for the node-to-client protocol Client types

### PR DESCRIPTION
**Pull request by dcoutts**
That is, for the wrapper types of the client side of all the
node-to-client protocols, specifically the types ChainSyncClient,
LocalStateQueryClient and LocalTxSubmissionClient.
    
These are useful for the cardano-api where we want to present the
protocol handlers using one set of types, but then execute them at the
proper wire types.

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Client.hs [hunk 1]
```diff
@@ -93,6 +98,62 @@ data ClientStQuerying block query m a result = ClientStQuerying {
       recvMsgResult :: result -> m (ClientStAcquired block query m a)
     }
 
+-- | Transform a 'LocalStateQueryClient' by mapping over the tx header and the
+-- chain tip values.
+--
+-- Note the mapping is contravariant in both types since they are outputs of
+-- the protocol, not inputs.
```
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Client.hs:105]: The comment is not right, it is covariant in block and query types and contravariant in the result type.

**Review by coot — APPROVED** [review]
LGTM, just one of the haddocs needs to be changed.

File: ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Client.hs (hunk 1 above)
Comment (dcoutts) [reply ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Client.hs]: Thanks. Yes.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**